### ISSUE TITLE
Column Visiblility with Bootstrap Integration and Column Reorder

### ISSUE THREAD

**Issue by MuellerMatthew** [DataTables/Buttons#26]
When working with Row Reorder and the bootstrap template integration the Column Visibility dropdown buttons break when a user reorders a column.

Before a user reorders a column, the buttons in the visibility dropdown are like:
`<li class="dt-button buttons-columnVisibility" tabindex="0" aria-controls="CD_Data"><a>Address</a></li>`
but after the columns are reordered the `<a>` tags around the 'column name' are missing on all of the buttons, so they are just plain text.

For reference I am using: 
Bootstrap 3.3.5, 
DataTables 1.10.9, 
Buttons 1.0.3, 
Column visibility 1.0.3, 
and
ColReorder 1.2.0

**Comment by MuellerMatthew** [comment]
Also, it seems like when setting column visibility to only a few columns, if you declare the columns it should be applied to by column name, when the columns are reordered the visibility does not move with the columns.  
Thus if I had 4 columns: column1, column2, column3, and column4, and I initialize my columns like: 
```{
        extend: 'colvis',
        columns:  'column1:name,'+
                  'column3:name,'+
                  'column4:name,'
      }
```
so that column2 can't have its visibility changed, if i reorder the columns, then the new column which is placed in the second column spot will no longer be on the colvis list, but 'column2' will now be there. because the colvis toggle is not attached to the column but it is attached to the columns position.

**Comment by hector** [comment]
+1

Exactly the same is happening to me and I arrived to the conclusion of changing the same line of code.

**Comment by hector** [comment]
If somebody wants a temporal patch for this problem I copy what I am using (base on @MuellerMatthew solution).
This can be executed anywhere once datatables, buttons and colvis scripts are loaded.

```javascript
  var original_init = $.fn.dataTable.ext.buttons.columnVisibility.init;
  $.fn.dataTable.ext.buttons.columnVisibility.init =
    function (dt, button, conf) {
      var that = this;
      original_init.call(this, dt, button, conf);
      dt.off('column-reorder.dt' + conf.namespace);
      dt.on('column-reorder.dt' + conf.namespace,
        function (e, settings, details) {
          var col = dt.column(conf.columns);
          var btn = button.children() ? button.children() : button;
          btn.text(conf._columnText(dt, conf.columns));
          that.active(col.visible());
        });
    };
```

**Comment by DataTables** [comment]
Fixed by 0ae182a6687690ec1361022ad93c167d9f8a1e3a and a7c0b6fbe0382f6039df438351bfb29dd8ed6743.
